src/data_api/reporting.se

```diff
@@ -27,186 +27,161 @@ extern branches: [addCurrency:[int256,int256,int256,int256]:int256, addMarketToB
 
 data controller
 
 def init():
     self.controller = 0x0
     REPCONTRACT = self.controller.lookup('repContract')
-    self.Reporting[1010101].repIDtoIndex[1010101] = 0
     redistributionRep = REPCONTRACT.balanceOf(0x0000000000000000000000000000000000000000)
-    self.Reporting[1010101].reputation[0].repValue = redistributionRep
-    self.Reporting[1010101].reputation[0].reporterID = 1010101
-    self.Reporting[1010101].dormantRep[0].reporterID = 1010101
-    self.Reporting[1010101].numberReporters = 2
-    self.Reporting[1010101].repIDtoIndex[msg.sender] = 1
-    self.Reporting[1010101].reputation[1].repValue = 0
-    self.Reporting[1010101].reputation[1].reporterID = msg.sender
-    self.Reporting[1010101].dormantRep[1].reporterID = msg.sender
-    self.Reporting[1010101].repIDtoIndex[msg.sender] = 1
-    self.Reporting[1010101].activeRep += redistributionRep
-    self.Reporting[1010101].totalRep += redistributionRep
-
-def getReportedOnNonFinalRoundTwoEvent(branch, reporter):
-    return(self.Reporting[branch].reportedOnNonFinalRoundTwoEvent[reporter])
-
-def setReportedOnNonFinalRoundTwoEvent(branch, event, reporter):
+    self.Reporting[1010101].activeRep[0x0000000000000000000000000000000000000000] = 0
+    self.Reporting[1010101].dormantRep[0x0000000000000000000000000000000000000000] = redistributionRep
+    self.Reporting[1010101].totalActiveRep = 0
+    self.Reporting[1010101].totalDormantRep = redistributionRep
+    # TODO: figure out why address 0 counts as a reporter, feels like this should be `0`
+    self.Reporting[1010101].numberReporters = 1
+    self.Reporting[1010101].reporterLocks[0x0000000000000000000000000000000000000000].eventCount = 0
+
+def getNumberOfEventsLockingReporter(branch, reporterAddress):
+    return(self.Reporting[branch].reporterLocks[reporterAddress].eventCount)
+
+def lockReporter(branch, reporterAddress, event):
     self.controller.checkWhitelist(msg.sender)
-    self.Reporting[branch].reportedOnNonFinalRoundTwoEvent[reporter] = event
+    self.Reporting[branch].reporterLocks[reporterAddress].events[event] = 1
+    self.Reporting[branch].reporterLocks[reporterAddress].eventCount += 1
     return(1)
 
-# @return fxp
-def getActiveRep(branch):
-    return(self.Reporting[branch].activeRep)
+def unlockReporter(branch, reporterAddress, event):
+    self.controller.checkWhitelist(msg.sender)
+    if (self.Reporting[branch].reporterLocks[reporterAddress].events[event] == 0):
+        throw()
+    if (self.Reporting[branch].reporterLocks[reporterAddress].eventCount <= 0):
+        throw() # if we are in this state things are _really_ broken
+    self.Reporting[branch].reporterLocks[reporterAddress].events[event] = 0
+    self.Reporting[branch].reporterLocks[reporterAddress].eventCount -= 1
+    return(1)
 
-def adjustActiveRep(branch, fxpAmount):
+def activateRep(branch, attorep):
```

Comment:
None of the setters in data_api are meant to be called by users. They're meant to be places to store + access state but have almost no business logic, which is done in the functional contracts. You'll notice this function, nor does the function that it calls that you wrote actually check whether a user's rep can be activated or not, that's done in the functions outside of this file that convertRepToDormant or convertRepToActive.  So no user should be able to call activateRep.  With that established, no _contract_ of ours should call it either, b/c then msg.sender would be the contract, not the user of the rep it meant to activate.  activateRepFor is thus the function that'd actually be used